### ISSUE TITLE
Calls to `provideConfigurations` should batch multiple URI's when possible

### ISSUE THREAD

**Issue by Colengms** [microsoft/vscode-cpptools#10468]
Currently, in `provideCustomConfiguration`, we're enumerating a list of URI's to provide configurations for, and requesting those configurations one at a time with separate calls to `provider.provideConfigurations`.  Coalescing those URIs into an array and making a single call to `provider.provideConfigurations` with all of those URIs would allow configuration providers to better optimize how configurations are gathered.

**Comment by Colengms** [comment]
Reopening.  https://github.com/microsoft/vscode-cpptools/issues/9003 did not fully address the issue.  Rather, it ensures that multiple results from `QueryTranslationUnitSource` (i.e. to find a source for a header file), were requested in a single query.

The rest of this fix would involve updating `provideCustomConfiguration` to accept an array of URI's, updating `provideCustomConfiguration` to batch up the results from `QueryTranslationUnitSource` to request everything at once, and passing in an array in `updateCustomConfigurations`.

Unfortunately, we don't need configs for _all_ of the results of `QueryTranslationUnitSource`.  We had been trying one at a time, since we don't know which, if any, associated source files might provide configurations, and we only need one.  So, requesting them all at once may actually create a lot of unnecessary work for the configuration provider, if all of those configurations are not easily available.  This might get addressed by Unification with https://github.com/microsoft/vscode-cpptools/issues/7526 , as Unification work requires that all configurations are available up-front.

**Comment by Colengms** [comment]
Closing this.  We've since switched to requesting individual custom configurations only on an as-needed basis, as TUs start up. 

While some configuration providers may benefit from bulk requests, that may incur overhead for other providers.  If a custom configuration provider may benefit from loading batches of configurations, it should probably manage that itself.  The C/C++ Extension doesn't have any additional information as to what files are likely to be needed later.

We do plan to eventually request configurations for all files at startup, which is required to enable certain feature as we unify the codebase with VC.
